src/common_util/poly.py

```diff
@@ -144,12 +144,48 @@ def __truediv__(self, other):
             if (self.basis == Basis.MONOMIAL):
                 c1 = self.values
                 c2 = [other]
-                res = P.polydiv(c1,c2)
+                qx, rx = P.polydiv(c1,c2)
+                assert rx == [0] # Divisor is a scalar, so remainder should be 0
                 return Polynomial(
-                    res,
+                    qx,
                     self.basis,
                 )
 
+    def __pow__(self, exponent):
+        assert isinstance(exponent, int) and exponent >= 0, "Exponent must be a non-negative integer"
+        
+        # Handle the simple cases
+        if exponent == 0:
+            # Any polynomial to the power of 0 is 1 (in monomial basis for simplicity)
+            return Polynomial([Scalar(1)], Basis.MONOMIAL)
+        elif exponent == 1:
+            return self
+        else:
+            # Repeated multiplication for exponentiation
+            result = self
+            for _ in range(exponent - 1):
+                result = result * self
+            return result
+
+    def div_with_remainder(self, other):
```

Comment:
This method shouldn't be used, all polynomial division we used should not have reminder